pennylane/transforms/defer_measurements.py

```diff
@@ -101,9 +113,309 @@ def null_postprocessing(results):
+            """
+            # pylint: disable=protected-access
+            # MeasurementValue._apply is for applying a new operation to the current
+            # MeasurementValue
+            # It is used when either:
+            # 1. A unary operation is performed on one MeasurementValue, e.g., ~m0
+            # 2. A binary operation is performed on a MeasurementValue and a scalar,
+            #    e.g., m0 + 1
+            #
+            # MeasurementValue._transform_bin_op is for applying a binary operation
+            # to two MeasurementValues
+            assert len(invals) <= 2
+            processing_fn = partial(eqn.primitive.bind, **eqn.params)
+
+            # One MeasurementValue
+            if len(invals) == 1:
+                m0 = invals[0]
+                return m0._apply(processing_fn)
+
+            # Two MeasurementValues
+            if all(isinstance(inval, MeasurementValue) for inval in invals):
+                m0, m1 = invals
+                return m0._transform_bin_op(processing_fn, m1)
+
+            # One MeasurementValue, one number
+            [m0, other] = invals if isinstance(invals[0], MeasurementValue) else invals[::-1]
+            return m0._apply(lambda x: processing_fn(x, other))
+
+        def eval(self, jaxpr: "jax.core.Jaxpr", consts: list, *args) -> list:
+            """Evaluate a jaxpr.
+
+            Args:
+                jaxpr (jax.core.Jaxpr): the jaxpr to evaluate
+                consts (list[TensorLike]): the constant variables for the jaxpr
+                *args (tuple[TensorLike]): The arguments for the jaxpr.
+
+            Returns:
+                list[TensorLike]: the results of the execution.
+
+            """
+            self._env = {}
+            self.setup()
+
+            for arg, invar in zip(args, jaxpr.invars, strict=True):
+                self._env[invar] = arg
+            for const, constvar in zip(consts, jaxpr.constvars, strict=True):
+                self._env[constvar] = const
+
+            for eqn in jaxpr.eqns:
+                primitive = eqn.primitive
+                custom_handler = self._primitive_registrations.get(primitive, None)
+
+                if custom_handler:
+                    invals = [self.read(invar) for invar in eqn.invars]
+                    outvals = custom_handler(self, *invals, **eqn.params)
+                elif getattr(primitive, "prim_type", "") == "operator":
+                    outvals = self.interpret_operation_eqn(eqn)
+                elif getattr(primitive, "prim_type", "") == "measurement":
+                    outvals = self.interpret_measurement_eqn(eqn)
+                else:
+                    invals = [self.read(invar) for invar in eqn.invars]
+                    if any(isinstance(inval, MeasurementValue) for inval in invals):
+                        outvals = self.resolve_mcm_values(eqn, invals)
+                    else:
+                        outvals = primitive.bind(*invals, **eqn.params)
```

Comment:
I am now more tempted to add a `interpreter_classical_operation` method to `PlxprInterpreter`. Should we just go ahead and do so?

Reply:
I think we can add multiple methods to abstract away details like this. The easier we can make it for everyone to avoid having to override `eval`, the better. Probably best to keep that in a separate PR and leave this as is for now.

Reply:
I added a shortcut story to track it [here](https://app.shortcut.com/xanaduai/story/84488).